pkg/crd/generator/generator.go

```diff
@@ -71,12 +71,16 @@ func (c *Generator) ValidateAndInitFields() error {
 		}
 	}
 
-	// Validate PROJECT file
-	if !crdutil.PathHasProjectFile(c.RootPath) {
-		return fmt.Errorf("PROJECT file missing in dir %s", c.RootPath)
+	// Validate PROJECT file if Domain or Repo are not set manually
+	if len(c.Domain) == 0 || len(c.Repo) == 0 {
```

Comment:
`crd` sub-command does not allow `Repo` field to be set so this condition always falls to requiring `PROJECT` file to exist:

```
$ bin/controller-gen crd --help
Generate CRD manifests from the Type definitions in Go source files.
Usage:
# controller-gen crd [--domain k8s.io] [--root-path input_dir] [--output-dir output_dir]

Usage:
  controller-gen crd [flags]

Flags:
      --apis-path string      the path to search for apis relative to the current directory (default "pkg/apis")
      --apis-pkg string       the absolute Go pkg name for current project's api pkg.
      --domain string         domain of the resources, will try to fetch it from PROJECT file if not specified
  -h, --help                  help for crd
      --namespace string      CRD namespace, treat it as cluster scoped if not set
      --output-dir string     output directory, default to 'config/crds' under root path
      --root-path string      working dir, must have PROJECT file under the path or parent path if domain not set
      --skip-map-validation   if set to true, skip generating OpenAPI validation schema for map type in CRD. (default true)
```

Is that an intent? Or, rather should it be `if len(c.Domain) == 0 && len(c.Repo) == 0 {` instead?